limitador-server/src/main.rs

```diff
@@ -276,10 +312,18 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                         // only reload when the limits file content changed
                         if location == last_known_canonical_path {
                             let limiter = limiter.clone();
+                            let status_updater = Arc::clone(&status_updater);
+
                             handle.spawn(async move {
                                 match limiter.load_limits_from_file(&location).await {
-                                    Ok(_) => info!("data modified; reloaded limit file"),
-                                    Err(e) => error!("Failed reloading limit file: {}", e),
+                                    Ok(_) => {
+                                        status_updater.write().unwrap().config_success();
```

Comment:
unwrap() here and below, not sure on the probability of status_updater.write() failing....
ah, not sure I follow why using write...but maybe me.

Reply:
Oh, and the `.write()` is because [`Status.config_success`](https://github.com/Kuadrant/limitador/pull/412/files#diff-77c4c3afe1e75f3f1602389f5895694a70395b1a7450ad7776945631b4bd28e9R223) requires a `&mut self`, as it mutates the struct. So we acquire the write lock of the `RwLock`